octavia_f5/controller/statusmanager/status_manager.py

```diff
@@ -32,6 +34,27 @@
 LOG = logging.getLogger(__name__)
 
 
+def rollback_on_db_exception(func):
+    """Call func and supply a session argument. If func raises an exception, roll back the session."""
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        lock_session = db_api.get_session(autocommit=False)
+        try:
+            kwargs['session'] = lock_session
+            return func(*args, **kwargs)
+        except db_exc.DBDeadlock:
+            LOG.debug('Database reports deadlock. Skipping.')
+            lock_session.rollback()
+        except db_exc.RetryRequest:
+            LOG.debug('Database is requesting a retry. Skipping.')
+            lock_session.rollback()
+        except Exception:
+            with excutils.save_and_reraise_exception():
+                lock_session.rollback()
+
+    return wrapper
+
+
```

Comment:
Please move the lock_session object generation outside of the wrapper function, it's strangly pops inside the function scope. Use instead of an function annotation the [with <wrapper>:](https://docs.python.org/2.5/whatsnew/pep-343.html) syntax.